# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Bailouts.h
# Commit: a5c379ff0fc7
# Full Hash: a5c379ff0fc7bc9d64a837bd3fb42dfcac5da10c
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2022-03-24 21:54:20
# Regressor Bug: 1760989
# File Overlap Count: 1
# Description:
#   Bug 1760989 part 5 - Factor out AssertBailoutStackDepth. r=iain
#   
#   We had similar stack depth assertions in two different places. This code is now
#   deduplicated and cleaned up a bit.
#   
# ==============================================================================

diff -r efd9e8d40b4c -r a5c379ff0fc7 js/src/jit/Bailouts.h
--- a/js/src/jit/Bailouts.h	Thu Mar 24 10:08:18 2022 +0000
+++ b/js/src/jit/Bailouts.h	Thu Mar 24 10:08:19 2022 +0000
@@ -225,6 +225,12 @@
 
 [[nodiscard]] bool FinishBailoutToBaseline(BaselineBailoutInfo* bailoutInfoArg);
 
+#ifdef DEBUG
+[[nodiscard]] bool AssertBailoutStackDepth(JSContext* cx, JSScript* script,
+                                           jsbytecode* pc, ResumeMode mode,
+                                           uint32_t exprStackSlots);
+#endif
+
 }  // namespace jit
 }  // namespace js
 